tests/test_encoder_decoder_models.py

```diff
@@ -0,0 +1,86 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+
+import torch
+from parameterized import parameterized
+from transformers import AutoModelForSeq2SeqLM
+
+from .testing_common import PeftCommonTester, PeftTestConfigManager
+
+
+PEFT_ENCODER_DECODER_MODELS_TO_TEST = [
+    "hf-internal-testing/tiny-random-T5ForConditionalGeneration",
+    "hf-internal-testing/tiny-random-BartForConditionalGeneration",
+]
+
+FULL_GRID = {
+    "model_ids": PEFT_ENCODER_DECODER_MODELS_TO_TEST,
+}
+
+
+def skip_non_lora(test_list):
+    return [t for t in test_list if "lora" in t[0]]
+
+
+class PeftEncoderDecoderModelTester(unittest.TestCase, PeftCommonTester):
+    r"""
+    Test if the PeftModel behaves as expected. This includes:
+    - test if the model has the expected methods
+
+    We use parametrized.expand for debugging purposes to test each model individually.
+    """
+    transformers_class = AutoModelForSeq2SeqLM
+
+    def prepare_inputs_for_testing(self):
+        input_ids = torch.tensor([[1, 1, 1], [1, 2, 1]]).to(self.torch_device)
+        decoder_input_ids = torch.tensor([[1, 1, 1], [1, 2, 1]]).to(self.torch_device)
+        attention_mask = torch.tensor([[1, 1, 1], [1, 0, 1]]).to(self.torch_device)
+
+        input_dict = {
+            "input_ids": input_ids,
+            "decoder_input_ids": decoder_input_ids,
+            "attention_mask": attention_mask,
+        }
+
+        return input_dict
+
+    @parameterized.expand(PeftTestConfigManager.get_grid_parameters(FULL_GRID))
+    def test_attributes_parametrized(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_model_attr(model_id, config_cls, config_kwargs)
+
+    @parameterized.expand(PeftTestConfigManager.get_grid_parameters(FULL_GRID))
+    def test_prepare_for_training_parametrized(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_prepare_for_training(model_id, config_cls, config_kwargs)
+
+    @parameterized.expand(PeftTestConfigManager.get_grid_parameters(FULL_GRID))
+    def test_save_pretrained(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_save_pretrained(model_id, config_cls, config_kwargs)
+
+    @parameterized.expand(
+        PeftTestConfigManager.get_grid_parameters(
+            {
+                "model_ids": PEFT_ENCODER_DECODER_MODELS_TO_TEST,
+                "lora_kwargs": {"init_lora_weights": [False], "merge_weights": [True, False]},
+            },
+        )
+    )
+    def test_merge_layers(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_merge_layers(model_id, config_cls, config_kwargs)
+
+    # skip non lora models - generate does not work for prefix tuning, prompt tuning
```

Comment:
`generate` should work for the prefix tuning. It won't work only for prompt tuning and p-tuning.

Reply:
Awesome let me dig into that!

Reply:
Appears the way I was constructing the tests was wrong, pushed a fix

Reply:
nit: update the comment to exclude prefix tuning?